tools/commands/build.py

```diff
@@ -66,37 +90,93 @@ def make(self):
         logger.info("building & installing {}: {}".format(self.name, " ".join(cmd)))
         subprocess.check_call(cmd, env=self.env)
 
+class acbuilder(builder):
+    def __init__(self, name, modules_dir, make_verbose=False):
+        self.make_verbose = make_verbose
+        super(acbuilder, self).__init__(name, modules_dir)
+
+    def clean(self):
+        if os.path.exists(self.build_path):
+            os.system("make -C {} uninstall".format(self.build_path))
+        super(acbuilder, self).clean()
+
+    def prepare(self):
+        os.chdir(self.src_path)
+        cmd = ["./build-tools/autogen.sh"]
+        logger.info("preparing {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
+        self.configure()
+
+    def configure(self):
+        os.mkdir(self.build_path)
+        os.chdir(self.build_path)
+        cmd = [self.src_path + "/configure",
+               "--prefix=" + self.install_path]
+        logger.info("configuring {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
+    def make(self):
+        cmd = ["make", '-C', self.build_path, "install"]
+        if self.make_verbose:
+            cmd.extend(["V=s"])
+        else:
+            cmd.extend(["-j", str(multiprocessing.cpu_count() + 1)])
+        logger.info("building & installing {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
 class mapbuild(object):
     def __init__(self, args):
         if args.verbose: logger.setLevel(logging.DEBUG)
+
         commands = args.commands
-        modules = build_modules if 'all' in args.modules else [m for m in build_modules if m in args.modules]
+        _map_modules = map_modules if 'all' in args.modules or 'map' in args.modules else [m for m in map_modules if m in args.modules]
+        _dep_modules = dep_modules if 'all' in args.modules or 'dep' in args.modules else [m for m in dep_modules if m in args.modules]
+
+        logger.info("{} {}".format(commands, _map_modules + _dep_modules))
+
```

Comment:
As mentioned above, I think install_dir and build_dir should be computed here and passed to all builder constructors.
```suggestion

build_dir = os.path.realpath(os.path.join(args.map_path), '..', 'build')
install_dir = os.path.join(build_dir, 'install')

```